### PR TITLE
Introduce the `meilitool` command line interface

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR introduces a small tool to help the Cloud team:
 - Clear the tasks queue by removing all the tasks
 - Dump a Meilisearch database without having to enqueue the task
 - Access this `meilitool` binary from the Docker Image

## TODO
 - [x] Modify the Docker File to ship with this new tool (@curquiza, could you review that, please?)
 - [x] Clear the tasks queue by removing all the tasks
   - [x] Add more logs to explain what is happening
   - [x] Clear the `update_files` folder
 - [x] Dump a Meilisearch database without having to enqueue the task
   - [x] Add more logs to explain what is happening
   - [x] Introduce a flag to skip dumping enqueued and processing tasks.
   - [x] Dump the instance uid.
   - [x] Dump the keys.
   - [x] Dump the tasks with the update files.
   - [x] Dump the index documents and settings.
   - [ ] ~Dump the experimental features~

### CODE REVIEW

**Comment by curquiza** [comment]
⚠️ @Kerollmops if you add the v1.5.0 milestones, please merge the PR into `release-v1.5.0` and not `main`

**Comment by curquiza** [comment]
@Kerollmops did you try to build a Docker prototype? Locally I get this error:

```
$ docker build -t test-meili-tool .

#12 248.3   Downloaded lindera-ko-dic v0.27.1
#12 248.7 error: package `anstream v0.6.4` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.69.0
#12 248.7 Either upgrade to rustc 1.70.0 or newer, or use
#12 248.7 cargo update -p anstream@0.6.4 --precise ver
#12 248.7 where `ver` is the latest version of `anstream` supporting rustc 1.69.0
------
executor failed running [/bin/sh -c set -eux;         apkArch="$(apk --print-arch)";         if [ "$apkArch" = "aarch64" ]; then             export JEMALLOC_SYS_WITH_LG_PAGE=16;         fi &&         cargo build --release -p meilisearch -p meilitool]: exit code: 101
```

**Comment by Kerollmops** [comment]
@curquiza, it seems that you should probably clear your Docker cache or something, as the current Rust version is 1.74 [according to this beautiful website](https://www.whatrustisit.com/) 😇

**Comment by curquiza** [comment]
@Kerollmops it does not work on the CI either -> https://github.com/meilisearch/meilisearch/actions/runs/6694669045/job/18188526604
⚠️ I recommend testing with the Docker CI directly (= create a prototype) to ensure we can release rc2 without issue.

**Review by curquiza — APPROVED** [review]
Just tested with a basic and healthy database (not a "stuck" one or anything) -> the tool is available and the commands work! 🎉

File: Dockerfile [hunk 1]
```diff
@@ -28,9 +28,10 @@ ENV     MEILI_SERVER_PROVIDER docker
 RUN     apk update --quiet \
         && apk add -q --no-cache libgcc tini curl
 
-# add meilisearch to the `/bin` so you can run it from anywhere and it's easy
-# to find.
-COPY    --from=compiler /meilisearch/target/release/meilisearch /bin/meilisearch
+# add meilisearch and meilitool to the `/bin` so you can run it from anywhere
+# and it's easy to find.
+COPY    --from=compiler /target/release/meilisearch /bin/meilisearch
+COPY    --from=compiler /target/release/meilitool /bin/meilitool
```
Comment (jlandic) [inline Dockerfile:34]: Looks all good for the Cloud too 👍 

Thanks for this! I'm sure it'll quickly prove useful 🚀

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**